pkg/services/object/put/service_test.go

```diff
@@ -0,0 +1,642 @@
+
+	// all
+	for _, member := range members {
+		assertObjectIntegrity(t, member)
+		require.LessOrEqual(t, member.PayloadSize(), limit)
+		require.Equal(t, sessionToken, *member.SessionToken())
+		require.Equal(t, sessionToken.Issuer(), member.Owner())
+		require.EqualValues(t, currentEpoch, member.CreationEpoch())
+		require.Empty(t, member.Attributes())
+		require.True(t, member.HasParent())
+	}
+
+	// payload chunks
+	chunks, linker := members[:len(members)-1], members[len(members)-1]
+
+	var gotPayload []byte
+	for _, chunk := range chunks {
+		require.Equal(t, object.TypeRegular, chunk.Type())
+
+		gotPayload = append(gotPayload, chunk.Payload()...)
+	}
+	require.Len(t, gotPayload, int(ln))
+
+	require.Zero(t, chunks[0].GetFirstID())
+	require.Zero(t, chunks[0].GetPreviousID())
+	for i := 1; i < len(chunks); i++ {
+		require.Equal(t, chunks[0].GetID(), chunks[i].GetFirstID())
+		require.Equal(t, chunks[i-1].GetID(), chunks[i].GetPreviousID())
+	}
+
+	// linker
+	require.Equal(t, object.TypeLink, linker.Type())
+	require.Equal(t, chunks[0].GetID(), linker.GetFirstID())
+	require.Zero(t, linker.GetPreviousID())
+
+	var link object.Link
+	require.NoError(t, link.Unmarshal(linker.Payload()))
+	linkItems := link.Objects()
+	require.Len(t, linkItems, len(chunks))
+	for i := range linkItems {
+		require.Equal(t, chunks[i].GetID(), linkItems[i].ObjectID())
+		require.EqualValues(t, chunks[i].PayloadSize(), linkItems[i].ObjectSize())
+	}
+
+	// parent
+	firstParent := chunks[0].Parent()
+	require.NotNil(t, firstParent)
+	require.Zero(t, firstParent.PayloadSize())
+	_, ok := firstParent.PayloadChecksum()
+	require.False(t, ok)
+	_, ok = firstParent.PayloadHomomorphicHash()
+	require.False(t, ok)
+	require.Zero(t, firstParent.GetID())
+	require.Zero(t, firstParent.Signature())
+
+	lastParent := chunks[len(chunks)-1].Parent()
+	require.NotNil(t, lastParent)
+	require.Equal(t, lastParent, linker.Parent())
+
+	for i := 1; i < len(chunks)-1; i++ {
+		require.Nil(t, chunks[i].Parent())
+	}
+
+	var firstParentCp object.Object
+	firstParent.CopyTo(&firstParentCp)
+	firstParentCp.SetPayloadSize(lastParent.PayloadSize())
+	if cs, ok := lastParent.PayloadChecksum(); ok {
+		firstParentCp.SetPayloadChecksum(cs)
+	}
+	if cs, ok := lastParent.PayloadHomomorphicHash(); ok {
+		firstParentCp.SetPayloadHomomorphicHash(cs)
+	}
+	firstParentCp.SetID(lastParent.GetID())
+	firstParentCp.SetSignature(lastParent.Signature())
+	require.Equal(t, firstParentCp, *lastParent)
+
+	restored := *lastParent
+	restored.SetPayload(gotPayload)
+
+	return restored
+}
+
+func splitMembersCount(limit, ln uint64) int {
+	if ln <= limit {
+		return 1
+	}
+	res := ln/limit + 1 // + LINK
+	if ln%limit != 0 {
+		res++
+	}
+	return int(res)
+}
+
+func assertObjectIntegrity(t *testing.T, obj object.Object) {
+	require.NoError(t, obj.CheckVerificationFields())
+
+	require.NotNil(t, obj.Version())
+	require.Equal(t, version.Current(), *obj.Version())
+
+	require.NotZero(t, obj.GetContainerID())
```

Comment:
Likely we know specific values we expect.

Reply:
this func checks general correctness, input-dependent checks are done where input is init